portal/src/it/scala/cromwell/pipeline/controller/ProjectControllerItTest.scala

```diff
@@ -58,12 +58,14 @@ class ProjectControllerItTest
         }
       }
 
-      "return status code 403 if user isn`t project owner" in {
-        val projectByNameRequest = dummyProject.name
-        val accessToken = AccessTokenContent(stranger.userId)
+      "return a status code 404 if project doesn't exist" in {
+
+        val strangerProject = TestProjectUtils.getDummyProject(ownerId = stranger.userId)
+        val projectByNameRequest = strangerProject.name
+        val accessToken = AccessTokenContent(dummyProject.ownerId)
 
         Get("/projects?name=" + projectByNameRequest) ~> projectController.route(accessToken) ~> check {
-          status shouldBe StatusCodes.Forbidden
+          status shouldBe StatusCodes.NotFound
```

Comment:
It's a bit strange that you need to modify a test that should be unrelated to your changes.

Reply:
It returns "NotFound" due to after changes we dont have case when method  getUserProjectByName may fail with ProjectAccessDeniedException, since we should return either project with correct name and userId or None, it seems to me.

Reply:
Previously we returned 403 if the project exists but the user doesn't have rights to see it.
How come that the ability to add projects with the same name affects it?
There are 3 cases:
1) There are no projects with such name
2) There are such projects, but the user can't see them
3) There is a project that user can see
What stops us from differentiating between these three cases apart from implementation details?